packages/revocation/src/contracts/RevocationRegistry.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >=0.7.0 <0.9.0;
+    
+contract RevocationRegistry {
+    
+    struct RevokedClaim {
+        address revoker;
+        uint revokedTimestamp;
+    }
+    mapping (bytes32 => mapping(address => RevokedClaim)) revokedClaims;
```

Comment:
I know this might seems a bit of reverse deja-vu from our previous conversation 😅 . But I think that we don't want to store the revocations in this way. I think it would be better to have a mapping to an array of `RevokedClaims`
```suggestion
    mapping (bytes32 => RevokedClaim[]) revokedClaims;
```
In this case, the bytes32 is not the role namehash but rather a hash of the credential or some data that identifies the credential.
This is for a few reasons:
- Unlike with on-chain role claims, privacy is important. So I think it should not be possible to figure out that a DID had a role by looking at its revocations.
- We need to have an array of revocations for a particular credential. We can't have a single one because we don't know if a particular revoker is an honest revoker (and so we don't want them to fill the single slot with a dishonest revocation).
- This more generic way means that the contract can be used for non-role credential revocations